OpenRA.Mods.Common/Activities/PickupUnit.cs

```diff
@@ -61,11 +62,13 @@ protected override void OnFirstRun(Actor self)
 				QueueChild(new Fly(self, Target.FromActor(cargo)));
 				QueueChild(new FlyIdle(self, idleTurn: false));
 			}
+			else
+				reserveFailed = true;
```

Comment:
reserve fails is the above `return` as well, tough as checks match it is caught in Tick. There shouldn't be any harm in setting `reserveFailed = true` in the above `return` as well

Reply:
Yea makes sense.